components/eam/src/dynamics/se/stepon.F90

```diff
@@ -417,6 +423,37 @@ subroutine stepon_run2(phys_state, phys_tend, dyn_in, dyn_out )
          call outfld('DIV',tmp_dyn(1,1,1,ie),npsq,ie)
       enddo
    endif
+
+   if (hist_fld_active('DYN_PNH')) then
+      do ie=1,nelemd
+         ! time level ntQ is not used
+         call get_field(dyn_in%elem(ie),'pnh',tmp_dyn(:,:,:,ie),hvcoord,tl_f,1)
+         call outfld('DYN_PNH',tmp_dyn(1,1,1,ie),npsq,ie)
+      enddo
+   endif
+
+   if (.not. theta_hydrostatic_mode) then
```

Comment:
Lets remove this check.  and remove this variable from the use statement above.  In hydrostatic mode, get_field_i() should do the correct thing for these variables (or abort if it can not).  that's preferable to silently not outputting them if the user requested them.